pkg/domain/infra/abi/images.go

```diff
@@ -806,3 +817,32 @@ func localPathFromURI(url *url.URL) (string, error) {
 	}
 	return url.Path, nil
 }
+
+func createSignature(m []byte, dockerReference string, mech signature.SigningMechanism, keyIdentity string) ([]byte, error) {
```

Comment:
Please use the original `reference.Named` instead of a `string` for `dockerReference`, to keep strong typing as long as possible.

(Absolutely non-blocking: Consider also passing all of `options` instead of `keyIdentity`. Either way works fine.)